src/app/backend/handler/apihandler.go

```diff
@@ -173,13 +171,38 @@ func FormatResponseLog(resp *restful.Response, req *restful.Request) string {
 		req.Request.RemoteAddr, resp.StatusCode())
 }
 
-// CreateHTTPAPIHandler creates a new HTTP handler that handles all requests to the API of the backend.
-func CreateHTTPAPIHandler(client *clientK8s.Clientset, heapsterClient client.HeapsterClient,
-	clientConfig clientcmd.ClientConfig) (http.Handler, error) {
+func (apiHandler *APIHandler) injectApiClient(fn func(*clientK8s.Clientset, *restful.Request, *restful.Response)) restful.RouteFunction {
+	return func(request *restful.Request, response *restful.Response) {
+		cfg, err := apiHandler.clientConfig.ClientConfig()
+		if err != nil {
+			handleInternalError(response, err)
+			return
+		}
+
+		token := request.HeaderParameter("Authorization")
+		if strings.HasPrefix(token, "Bearer ") {
```

Comment:
I exepected https://golang.org/pkg/strings/#HasPrefix to be case sensitive so I'm not sure what you're trying to tell me.